src/odemis/acq/test/path_test.py

```diff
@@ -1184,6 +1188,133 @@ def test_set_path_stream(self):
+
+    def test_set_path(self):
+        """
+        Test setting light-in-align
+        """
+        # Light aligner should always be in active position, for any mode
+
+        # Start by going to a different mode: use ar (lens 2 is active)
+        self.optmngr.setPath("ar").result()
+        # Assert that actuator was moved according to mode given
+        assert_pos_as_in_mode(self, self.lens_switch, "ar")
+        assert_pos_as_in_mode(self, self.slit, "ar")
+        assert_pos_as_in_mode(self, self.spectrograph, "ar")
+        assert_pos_as_in_mode(self, self.light_align, "ar")
+        self.assertEqual(self.spec_det_sel.position.value, {'rx': 0})
+
+        # setting light-align: lens 2 should be deactive
+        self.optmngr.setPath("light-in-align").result()
+        # Assert that actuator was moved according to mode given
+        assert_pos_as_in_mode(self, self.lens_switch, "light-in-align")
+        assert_pos_as_in_mode(self, self.slit, "light-in-align")
+        assert_pos_as_in_mode(self, self.light_align, "light-in-align")
+
+
+class Sparc2SpecSwitchTestCase(unittest.TestCase):
+    """
+    Tests of a (simulated) SPARC2 with a spec-switch component (FSLM)
+    """
+    @classmethod
+    def setUpClass(cls):
+        testing.start_backend(SPARC2_FSLM_CONFIG)
+
+        # Microscope component
+        cls.microscope = model.getComponent(role="sparc2")
+        # Find CCD & SEM components
+        cls.ccd = model.getComponent(role="ccd")
+        cls.spectrograph = model.getComponent(role="spectrograph")
+        cls.lens_switch = model.getComponent(role="lens-switch")
+        cls.lens_mover = model.getComponent(role="lens-mover")
+        cls.spec_switch = model.getComponent(role="spec-switch")
+        cls.slit = model.getComponent(role="slit-in-big")
+        cls.light_align = model.getComponent(role="light-aligner")
+        cls.opt_mngr = path.OpticalPathManager(cls.microscope)
+
+        # add spec-switch attribute to light-in-align mode just for testing. normally it is left out of the
+        # modes because we do not want to automatically switch it back and forth while switching modes
+        # cls.opt_mngr._modes["light-in-align"][1].update({'spec-switch': {'x': "MD:" + model.MD_FAV_POS_DEACTIVE}})
+
+    @classmethod
+    def tearDownClass(cls):
+        del cls.opt_mngr  # garbage collect it
+
+    def test_spec_switch_mirror(self):
+        """
+        Default movement of the spec-switch mirror is tested.
+        Basically the mirror moves either to engage position or to retract position.
+        The moves however are executed through a progressive future. This future type is
+        represented in odemis.util.driver as a generic class.
+        """
+        spec_sw_md = self.spec_switch.getMetadata()
+
+        # set the spec-switch mirror to the default retracted position (FAV_POS_DEACTIVE)
+        prog_f = ProgressiveMove(self.spec_switch, spec_sw_md[model.MD_FAV_POS_DEACTIVE])
```

Comment:
Please don't mix up the test cases. You can move here just with `self.spec_switch.moveAbs()`. The ProgressiveMove class should be tested, but in odemis/util/test/driver_test.py

You can use the `stage` component of the SECOM_CONFIG to test the class. In particular, it'll be easier to test with multiple axes.